### PR TITLE
chore: Bump qlog to 0.13 and deal with the fallout

### LINKED ISSUE #1826: Bump qlog to 0.13
It's out: https://github.com/cloudflare/quiche/commit/00638b167684b7631feef4cbe71dff904bb097cf

That should also enable us logging padding length.

### PR DESCRIPTION
**Pull request by larseggert**
Also make `log` and `qlog` workspace dependencies.

Fixes #1826

### CODE REVIEW

File: neqo-bin/Cargo.toml [hunk 1]
```diff
@@ -34,7 +34,7 @@ neqo-crypto = { path = "./../neqo-crypto" }
 neqo-http3 = { path = "./../neqo-http3" }
 neqo-qpack = { path = "./../neqo-qpack" }
 neqo-transport = { path = "./../neqo-transport" }
-qlog = { version = "0.12", default-features = false }
+qlog = { version = "0.13", default-features = false }
```
Comment (mxinden) [inline neqo-bin/Cargo.toml:37]: How about promoting `qlog` to a workspace dependency? Thus a version bump would have to touch a single file only.

https://doc.rust-lang.org/cargo/reference/workspaces.html#the-dependencies-table

File: neqo-transport/src/qlog.rs [hunk 2]
```diff
@@ -392,8 +392,14 @@ impl From<&Frame<'_>> for QuicFrame {
         match frame {
             // TODO: Add payload length to `QuicFrame::Padding` once
             // https://github.com/cloudflare/quiche/pull/1745 is available via the qlog crate.
```
Comment (mxinden) [inline neqo-transport/src/qlog.rs:394]:
```suggestion
```

Outdated now, right?

**Review by mxinden — APPROVED** [review]

**Review by KershawChang — APPROVED** [review]

File: neqo-bin/Cargo.toml (hunk 1 above)
Comment (larseggert) [reply neqo-bin/Cargo.toml:37]: Is there a downside if not every workspace member actually needs the dependency? (Such as `fuzz`.)

Comment (mxinden) [reply neqo-bin/Cargo.toml:37]: I am not aware of one @larseggert. We have been using them quite extensively in https://github.com/libp2p/rust-libp2p/blob/bee8199c0db79e2b621e3a076ed09ece65337553/Cargo.toml#L73-L127.

**Review by mxinden — APPROVED** [review]
:+1:

**Review by KershawChang — APPROVED** [review]

**Pull request merged**